krabs/krabs/parse_types.hpp

```diff
@@ -253,6 +253,42 @@ namespace krabs {
     private:
     };
 
+    /**
+    * <summary>
+    * Used to handle parsing of POINTER types
+    */
+    struct pointer {
+        // Make the reference const and call it wild,
+        // As any attempt to derefernce or alter what it points to
+        // would be very undefined behaviour, as it
+        // is unlikley it is pointing to inside our process' memory
+        const void* wild_reference;
```

Comment:
Kind of a strange name. Few thoughts: 1) why even store it as an actual pointer at all? It's probably never a good idea to use it as a pointer in code. You could just keep a uint64_t value and let the top bytes be empty on x86. Just call it value or something. Alternately, you could make wrap this in a a type that doesn't allow dereferencing at all (just overload operator* and make it assert, but ultimately you'll still want to see the value at some point, so it seems like using an integer may be better

If you really want a pointer (so things like pointer printf specifiers work), I'd still suggest calling it `value` instead of `wild_reference`.

Reply:
Hey Pat,

Nice work.  :-)

I think that using a uint64_t should be enough of a warning to the developer.
And perhaps just 'address' if it needs a name?

John

For reference -
perfview's TraceEvent defines an 'Address' type which is just a UInt64.
And the default string representation is 0x%x.

https://github.com/microsoft/perfview/blob/9d8a9b90c5ed917da1b9bdea98210114a270e633/src/TraceEvent/TraceEvent.cs#L20
https://github.com/microsoft/perfview/blob/9d8a9b90c5ed917da1b9bdea98210114a270e633/src/TraceEvent/TraceEvent.cs#L978